### DISCUSSION TITLE
Sveltekit + Prisma in node release error 500

### DISCUSSION THREAD

**Discussion by iBuaSpaz** [prisma/prisma#20033] [Q&A] [answered]
I tried to search for a answer to this,

After i build my sveltekit app in node mode, and uploaded it to my aaPanel host as a node project, i get error 500 whenever it need to take items from mysql database.

My .env have been changed to the new user/password for mysql in project folder.

After build the build folder doesnt seem to include the .env file as well.

All tutorials just have it running in the dev enviroment, none really say how to build this and deploy it.

Am i missing a step? where does the build node application store the mysql string?

**Comment by nurul3101** [comment]
Hello @iBuaSpaz 👋 

Could you please share the entire error message that you get?
How are you building your project? Which command are you using?
Can you please share the tutorial that you are referring to while creating your application?

**Reply by iBuaSpaz** [reply, marked as answer]
I found sollution, aaPanel was just acting wierd, the tutorials are just those on youtube and other sites, not many actually have a release tutorial, like for me using node index.js kept crashing (because some 404 images), but the pm2 index.js works like a charm.

**Comment by ludralph** [comment]
Hi @iBuaSpaz 👋 

When you build a SvelteKit app in Node mode, the `.env` file is not automatically included in the build output. The `.env` file is typically used for local development and should not be deployed to a production environment.

Instead, you need to configure the environment variables on your hosting server or hosting provider's control panel. In your case, with `aaPanel`, you'll need to set the environment variables for your MySQL connection there.

To set environment variables in `aaPanel`, you can follow these steps:

1. Log in to your `aaPanel` dashboard.
2. Navigate to the `Web` section or the specific website you want to configure.
3. Look for the `Environment Variables` or `Variables` section.
4. Add a new environment variable for your MySQL connection, such as `DB_CONNECTION_STRING`, and set its value to your MySQL connection string.

By setting the environment variable in your hosting server's control panel, you can access it within your SvelteKit app using import.meta.env.VARIABLE_NAME. For example, if you set DB_CONNECTION_STRING as the environment variable, you can access it in your SvelteKit app as import.meta.env.VITE_DB_CONNECTION_STRING.

Make sure to update your SvelteKit app code to use the environment variable to establish a connection with your MySQL database.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by iBuaSpaz** [comment]
I found a solution thanks :)

Sadly aaPanel does not have a web section nor a enviroment variables section, a friend who does web developement came by and had a look.
